src/python/pants/backend/adhoc/run_system_binary.py

```diff
@@ -87,6 +92,9 @@ async def _find_binary(
         env.update(**(rds.extra_env or {}))
         append_only_caches = rds.append_only_caches
         immutable_input_digests = rds.immutable_input_digests
+    if extra_env_vars:
+        extra_env = await Get(EnvironmentVars, EnvironmentVarsRequest(extra_env_vars))
+        env.update(extra_env)
```

Comment:
This could stomp the PATH or _PANTS_SHIM_ROOT, no? Seems like the update order should be reversed.

Reply:
I reversed the order